<h1>ValidPrime Help</h1>
<h3>ValidPrime has 3 objectives:</h3>
<ol>
<li>
It tests the gDNA sensitivity of qPCR assays and 
classifies the assays into gDNA-insensitive and gDNA-sensitive assays.
</li>
<li>
For the gDNA-sensitive assays, the gDNA contribution in 
each individual sample is calculated and the input signal (CqNA)
is corrected accordingly to generate CqRNA, 
which corresponds to the signal derived from RNA.
</li>
<li>
ValidPrime replaces traditional RT minus controls,
while requiring fewer control reactions.
</li>
</ol>

<p> To run a ValidPrime experiment, 2 components are added 
to the traditional RT-qPCR setup:</p>
<ol>
<li>
A gDNA specific ValidPrime assay (VPA) targeting
transcriptionally silent gDNA
</li>
<li>
Purified gDNA sample(s) from the same species as the other samples
</li>
</ol>

<p>
For gDNA-sensitive assays, the accuracy of the correction depends
on the efficiency by which the Gene-of-interest (GOI)-assays amplify gDNA.
The best result is obtained using assays with efficiencies identical
to that of the VPA.
It is possible to evaluate this parameter in the ValidPrime software
(see ValidPrime assay validation below).
</p>

<h3>ValidPrime consists of 4 steps that must be followed in order:</h3>
<p>
The order is shown in the top toolbar,
<a href="#load">Load</a>
 -> 
<a href="#check">Check</a>
 -> 
<a href="#run">Run</a>
 -> 
<a href="#save">Save</a> Files
.
</p>
<p>
Elements of the interface that are not a part of the current step will be
inactive to aid you in following the steps.
You can use the Reload button to return to the load step,
and the Recheck button to return to the configuration step.
</p>

<p><i>Notes specific to Version 0.9.5</i></p>
<p>
For demonstration purposes, a small example file is included in the program.
Use the File Menu to Load the Demo input.
Then proceed to Check and Run to see how ValidPrime works. 
The demo data comes from an experiment where exogenous gDNA
was spiked into mouse kidney cDNA.
Another example file in BioMark format is available for download
at http://code.google.com/p/gh-validprime/downloads/list.
</p>


<h2><a name="load">Loading Input</a></h2>
<p>
Input files must be in Stepone or BioMark/heatmap format
(currently heatmap format needs to be tab-delimited text (.txt) files
with a '.' (not a ',') as the decimal point).
Spreadsheet format (also .txt), typically output from excel-type programs,
is also supported.
Spreadsheet files should have the Assay names as a header and 
the Sample names in the first column.
</p>
<p>
The loaded file data will appear in the window on the right.
The first row should show the assay names and 
the first column should indicate the sample names.
</p>
<p>
If you try to load a file in a format different from the format specified
in the Input Format field,
the program may crash or you will end up with garbage in the grid,
so be sure to check your input in the grid before proceeding.
</p>

<p><i>Notes specific to Version 0.9.5</i></p>
<p>
Stepone format contains variations that depend upon the version
of the upstream software.
For now, ValidPrime may or may not work on your file.
This will be fixed by the 1.0.1 release.
</p>

<h2><a name="check">Checking Input/Parameters</a></h2>
<p>
The Validate Assays checkbox should only be checked if you are testing
your qPCR primers for the first time with ValidPrime.
A gDNA dilution series with at least 4, 
preferably more samples/concentrations should be included.
This serves 2 purposes: 
</p>
<ul>
<li>
First, it allows to confidently identifying gDNA-insensitive assays
(see ValidPrime Scoring below).
</li>
<li>
Second, it allows an approximate estimation of the gDNA amplification 
efficiency. 
ValidPrime can handle replicates of dilutions series as well as replicates
of one gDNA concentration.
However, in the latter case, no qPCR efficiencies will be estimated
since ValidPrime only calculates these based on Cq values from dilution series.
The dilution series should typically span three log10,
(eg. 10, 50, 250, 1000, 5000 haploid genome copy numbers). 
Any gDNA-sensitive assay should readily amplify 10 copies. 
The sizes of most sequenced genomes (in basepairs) are available at
ensemble (http://www.ensembl.org, section: Browse a genome; Genome Statistics).
A copy number calculator is available at
https://www.finnzymes.fi/java_applets/copy_number_calculation.html.
Alternatively, the genome size database
(http://www.genomesize.com)
contains similar information for a large number of organisms,
expressed as C-values (pg/haploid genome).
The information in this database is not necessarily based on sequence data.
</li>
</ul>

<p>
The Cq cutoff is used to flag input values above a threshold
(flag : LOWSIGNAL).
This parameter should be used with caution since the limit of detection
is assay-dependent and the value will effect the number of A+ GOIs
(see ValidPrime Scoring below).
The default value is set to 40.
</p>
<p>
The user can manually flag cells in spreadsheets.
NOAMP refers to cells with no amplification detected.
EXPFAIL refers to cells in which a signal is detected but
the quality of the amplification is considered inadequate.
The Pass/Fail grid in the BioMark heatmap output is read 
and interpreted as follows:
</p>

<table>
<tr>
	<td>Cq</td>
	<td>BioMark Flag</td>
	<td>VP Flag</td>
	<td>Comment</td>
</tr>
<tr>
	<td>Below Cutoff</td>
	<td>Pass</td>
	<td>CALC</td>
	<td>Data is used in calculations</td>
</tr>
<tr>
	<td>Below Cutoff</td>
	<td>Fail</td>
	<td>EXPAIL</td>
	<td>Data is not used</td>
</tr>
<tr>
	<td>Above cutoff</td>
	<td>Pass/Fail</td>
	<td>LOWSIGNAL</td>
	<td>Data is not used</td>
</tr>
<tr>
	<td>999</td>
	<td>Pass/Fail</td>
	<td>NOAMP</td>
	<td>No data available</td>
</tr>
</table>

<p>
A similar Pass/Fail grid analysis for spreadsheet inputs
is not implemented in the present version of ValidPrime.
</p>
<p>
In order to run ValidPrime,
certain row and column labels must be selected to dictate
where the gDNA Sample columns are and where the ValidPrime Assay
is in the input.
Input must be loaded prior to the Check step so that the row and column 
labels are available within the program.
</p>
<p>
If the gDNA concentrations are not the sole numeric entities within 
the selected row labels,
 uncheck the "Parse gDNA concentrations from row labels?" checkbox 
and enter them manually using the Check/Set [gDNA] button.
</p>
<p>
The Ignore widgets allow the exclusion of specified assays (columns)
and/or samples (rows) from the analysis. 
</p>

<h2>ValidPrime assay validation</h2>
<p>
Based on the results from the gDNA dilution series,
ValidPrime evaluates whether or not CqRNA calculation is recommended
for a given assay.
Minimal number of valid cells/assay in the gDNA dilution series required
for High Confidence evaluation (see below).  
</p>
<p>
If a GOI assay targets gDNA, qPCR against gDNA at a given concentration 
will generate a Cq value which is either above or below the Cq obtained
with the VPA.
The difference between the two Cq will generate a delta Cq.
If the efficiency of the GOI-gDNA amplification is similar to the one obtained
with the VPA,
the delta Cq will vary little over the dilution series.
ValidPrime uses the standard deviation (SD) cutoff of the delta Cq values
to identify high confidence assays for which a ValidPrime adjustment is 
recommended.
The default All-SD Cutoff is set to 0.3 Cq and refers to if all
concentrations in the standard curve are taken into account.
As an option, ValidPrime can also identify aberrant results and exclude
the concentration which contributes the most to the dispersion in the data.
This refers to the LOO (Leave-One-Out) option (not activated by default).
LOO could be used if one of the gDNA concentrations appears to behave 
as an outlier.
The minimal required number of gDNA concentrations before LOO is 5. 
</p>

<p>
It is important to note that assay validation in ValidPrime does not
replace the standard criteria for assay validation according to the 
MIQE guidelines (http://www.rdml.org/miqe.php).
The purpose for assay validation in ValidPrime is merely to verify that
GOI assays, gDNA-sensitive or not, behave as expected against a gDNA template
and that the amplification efficiency for gDNA-sensitive assays is similar 
to that of the VPA.
</p>

When using a gDNA dilution series, if an Assay has at least Min-Count for SD values, the standard deviation will be calculated and checked against High All-SD Cutoff to mark High-Confidence Assays.
</p>

<h2> ValidPrime Scoring</h2>
<p>
ValidPrime assigns Grades based on the %DNA in each cell of the input grid.
The %DNA is defined as the fraction of the total signal (expressed in %) 
that is derived from gDNA.
The defaults are 3% or less for an A, 25% or less for a B, and 
60% or less for a C.
Any cell in which the %-DNA is greater than or equal to the Grade C max
will be flagged as HIGHDNA.
Assays that do not amplify gDNA are assigned A+/A++ or A+++,
depending of the confidence level of the test.
An assay which do not amplify gDNA at any concentration over an entire
dilution series get a higher confidence score than an assay that is tested 
on only one gDNA concentration.
The following weight matrix is used to define the Confidence Score.
</p>
<table>
<tr>
<td>Flag</td>
<td>Score</td>
</tr>
<tr>
<td>NOAMP</td>
<td>2 points</td>
</tr>
<tr>
<td>LOWSIGNAL</td>
<td>1 point</td>
</tr>
<tr>  
<td>EXPFAIL</td>
<td>0 point</td>
</tr>
<tr>
<td>CALC</td>
<td>-1 point</td>
</tr>
</table>
<p>
The minimal score for an A+++ assay is 8 points,
A++ 5 points and A+ 2 points.
To minimize the risk of false positives in designation of gDNA-insensitive
A+ assays, ValidPrime gives only 1 point a to NOAMP results
in gDNA samples with less than 20 genome copies. 
Whereas A+++ assays normally can be considered as DNA-insensitive,
we recommend a second validation of A+ and A++ assays.
</p>

<p>
In the case the VPA generates a NOAMP or a LOWSIGNAL result 
in a sample (meaning that the level of gDNA contamination is below 
the level detection or the Cq value was above the cutoff),
ValidPrime will add one Cq to the Cq cutoff value and use this 
in the calculations. These samples are labelled A*.
</p>

<h2><a name="run">Run ValidPrime</a></h2>
<p>
If Check was successful, the Run button will become enabled
and you can run ValidPrime.
</p>

<h2><a name="save">Viewing and Saving Results</a></h2>
<p>
In the Output sheet, the desired localization for saved result files 
is chosen with the output folder button. The Generate Output Names 
button can be used to create a series of output file names based on
the input file. 
</p>
<p>
Select any output files you want to create via the individual checkboxes.
Upon clicking 'Save' each selected output file will be created.
The files are saved in .txt with can be opened with Excel
among other programs.
</p>
<p>
In order to merely view the results within ValidPrime,
you can use the View menu to select the results to view in a simple grid.
The Toggle Grid button can be used to show different types of values
in the main application grid as well.
</p>
<p>
The ValidPrime scores can also be viewed in a HeatMap 
with custom coloring-schemes.
</p>
